queue/priority_queue.go

```diff
@@ -253,6 +258,12 @@ func (pq *PriorityQueue) Dispose() {
 	pq.waiters = nil
 }
 
+// AllowDuplicates determines whether the queue supports
+// duplicated items. This setting is false by default.
+func (pq *PriorityQueue) AllowDuplicates(allow bool) {
+	pq.allowDuplicates = allow
```

Comment:
The constructor approach, although introducing a compromise for existing users, looks like a simpler solution overall. The flag should not be changed after instantiation IMO.